modules/social_features/social_core/social_core.module

```diff
@@ -512,10 +512,23 @@ function social_core_filter_process_format($element) {
     $element_id = substr($element_id, 0, $sub_id);
   }
 
-  $full_html = FilterFormat::load('full_html');
-  $permission_name = $full_html->getPermissionName();
-  $account = \Drupal::currentUser();
-  if ($element['#type'] == 'text_format' && $element['#format'] == 'full_html' && !$account->hasPermission($permission_name)) {
+  // Default filter format for text area fields.
+  $social_filter_format = 'full_html';
+
+  // Allow module to override default filter format because some content type
+  // could require specific one.
+  Drupal::moduleHandler()->alter('social_filter_format_default', $social_filter_format);
+
+  /* @var \Drupal\filter\Entity\FilterFormat $social_filter_format */
+  $social_filter_format = FilterFormat::load($social_filter_format);
```

Comment:
I think we need to have an additional check here if filter format was loaded, to prevent fatal errors if wrong value was provided from hook.

Reply:
Added checks bellow